src/main/scala/tech/cryptonomic/conseil/tezos/ApiOperations.scala

```diff
@@ -738,49 +597,33 @@ object ApiOperations {
         val joinedTables = getJoinedTables(blockFlag, operationGroupFlag, operationFlag, accountFlag, filteredTables, filter)
         val action = joinedTables match {
 
-          case Some(OperationGroups(operationGroups)) =>
-            for {
-              opGroup <- operationGroups
-            } yield extractFromOperationGroup(opGroup)
+          case Some(OperationGroups(operationGroups)) => operationGroups
 
           case Some(BlocksOperationGroups(blocksOperationGroups)) =>
-            for {
-              (_, opGroup) <- blocksOperationGroups
-            } yield extractFromOperationGroup(opGroup)
+            blocksOperationGroups.map(_._2)
 
           case Some(OperationGroupsOperations(operationGroupsOperations)) =>
-            for {
-              (opGroup, _) <- operationGroupsOperations
-            } yield extractFromOperationGroup(opGroup)
+            operationGroupsOperations.map(_._1)
 
           case Some(OperationGroupsAccounts(operationGroupsAccounts)) =>
-            for {
-              (opGroup, _) <- operationGroupsAccounts
-            } yield extractFromOperationGroup(opGroup)
+            operationGroupsAccounts.map(_._1)
 
           case Some(OperationGroupsOperationsAccounts(operationGroupsOperationsAccounts)) =>
-            for {
-              ((opGroup, _), _) <- operationGroupsOperationsAccounts
-            } yield extractFromOperationGroup(opGroup)
+            operationGroupsOperationsAccounts.map(_._1)
 
           case Some(BlocksOperationGroupsOperations(blocksOperationGroupsOperations)) =>
-            for {
-              ((_, opGroup), _) <- blocksOperationGroupsOperations
-            } yield extractFromOperationGroup(opGroup)
+            blocksOperationGroupsOperations.map(_._2)
 
           case Some(BlocksOperationGroupsOperationsAccounts(blocksOperationGroupsOperationsAccounts)) =>
-            for {
-              (((_, opGroup), _), _) <- blocksOperationGroupsOperationsAccounts
-            } yield  extractFromOperationGroup(opGroup)
+            blocksOperationGroupsOperationsAccounts.map(_._2)
 
           case _ =>
-            throw new Exception("This exception should never be reached, but is included for completeness.")
+            throw new IllegalStateException("This exception should never be reached, but is included for completeness.")
         }
 
         val OperationGroupsAction(sortedAction) = fetchSortedAction(filter.order, OperationGroupsAction(action), filter.sortBy)
         val op = dbHandle.run(sortedAction.distinct.take(getFilterLimit(filter)).result)
-        val results = Await.result(op, Duration.apply(awaitTimeInSeconds, SECONDS))
-        results.map(x => Tables.OperationGroupsRow(x._1, x._2, x._3, x._4, x._5, x._6))
+        Await.result(op, awaitTimeInSeconds.seconds)
```

Comment:
Is the mapping irrelevant now because  the return type of results is now a database row, thanks to the mapping in the pattern matching plus your changing of the Joined Tables type signatures?

Reply:
Yes, if I got your comment correctly.

The JoinedTables now maps to tuples of xxxRow results, and once you get the table-specific row type out of the tuple, that has already the matching fields needed